promotion/datastore.go

```diff
@@ -1804,6 +1806,46 @@ func (pg *Postgres) RunNextDrainRetryJob(ctx context.Context, worker DrainRetryW
 	}
 }
 
+// RunNextFetchRewardGrantsJob - runs fetch reward grant jobs
+func (pg *Postgres) RunNextFetchRewardGrantsJob(ctx context.Context, worker SwapRewardsWorker) error {
+
+	service, _ := InitService(ctx, pg, nil, nil)
```

Comment:
Not sure we want to InitService again?  I think InitService is really only meant to be run once on service startup.

Reply:
I wasn't sure of a better way to go from `pg` and `ctx` to `service` that was needed further down.

Reply:
https://github.com/brave-intl/bat-go/pull/1233/commits/dc36d86c2d9ba0e0190d72478e3b5af177bef05d